internal/app/cf-terraforming/cmd/generate.go

```diff
@@ -1068,6 +1068,23 @@ func generateResources() func(cmd *cobra.Command, args []string) {
 				// tls_1_2_only is deprecated in favour of min_tls
 				jsonStructData[i].(map[string]interface{})["settings"].(map[string]interface{})["tls_1_2_only"] = nil
 			}
+		case "cloudflare_tiered_cache":
+			tieredCache, err := api.GetTieredCache(context.Background(), &cloudflare.ResourceContainer{Identifier: zoneID})
+			if err != nil {
+				log.Fatal(err)
+			}
+			var jsonPayload []cloudflare.TieredCache
+			jsonPayload = append(jsonPayload, tieredCache)
+
+			resourceCount = 1
+			m, _ := json.Marshal(jsonPayload)
+			err = json.Unmarshal(m, &jsonStructData)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			jsonStructData[0].(map[string]interface{})["id"] = zoneID
+			jsonStructData[0].(map[string]interface{})["cache_type"] = tieredCache.Type.String()
```

Comment:
why does this need remapping? isn't this the value in the schema already?

Reply:
Heya @jacobbednarz thanks for the feedback! Correct me if I'm wrong, I was trying to follow the schema defined in the provider here https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/tiered_cache

```
resource "cloudflare_tiered_cache" "example" {
  zone_id    = "0da42c8d2132a9ddaf714f9e7c920711"
  cache_type = "smart"
}
```
while the result from the API is type `TieredCache` as following, the `TieredCacheType` is enum so I suppose we need to call `String()` for translation. 
```
type TieredCache struct {
	Type         TieredCacheType
	LastModified time.Time
}
```
If there is a better way, kindly tell me and will update right away.

Reply:
that is all 👍 in the past we used to call `String()` automatically if it existed but i forgot we ditched it. cheers for the explanation.